modules/api/src/main/java/org/eclipse/xpanse/api/ServiceRegisterApi.java

```diff
@@ -200,15 +200,15 @@ public Response unregister(
     @GetMapping(value = "/services/register", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     public List<RegisteredServiceVo> listRegisteredServices(
-            @Parameter(name = "categoryName", description = "category of the service")
-            @RequestParam(name = "categoryName", required = false) Category category,
+            @Parameter(name = "categoryName", description = "name of category of the service")
+            @RequestParam(name = "categoryName", required = false) String categoryName,
```

Comment:
category name must be an enum everywhere.

Reply:
Using Category  with @RequestParam, when using error category will return http code 500, this is is a conflict with PR #672

Reply:
@baixinsui error code 422 is fine. We definitely need this enum to work insted of string.

Reply:
Using enum object instead of string will return 500 which handled by Swagger not 422,  I have not found other related solutions.

Reply:
I added a new exception handler to handle the MethodArgumentTypeMisMatchException and return 422 not 500